_schemaV2.graphql

```diff
@@ -14117,6 +14117,51 @@ type Query {
   ): ViewingRoomConnection @deprecated(reason: "Use viewingRoomsConnection")
 }
 
+type Quiz {
+  completedAt: String
+
+  # A globally unique ID.
+  id: ID!
+
+  # A type-specific ID likely used as a database ID.
+  internalID: ID!
+  quizArtworks(
+    after: String
+    before: String
+    first: Int
+    last: Int
+    page: Int
+    size: Int
+  ): QuizArtworkConnection
+}
+
+type QuizArtwork {
+  artwork: Artwork!
+  artworkId: String!
+  interactedAt: String
+  position: Int!
+}
+
+# A connection to a list of items.
+type QuizArtworkConnection {
+  # A list of edges.
+  edges: [QuizArtworkEdge]
+  pageCursors: PageCursors!
+
+  # Information to aid in pagination.
+  pageInfo: PageInfo!
+  totalCount: Int
+}
+
+# An edge in a connection.
+type QuizArtworkEdge {
+  # A cursor for use in pagination
+  cursor: String!
+
+  # The item at the end of the edge
+  node: QuizArtwork
```

Comment:
You almost never want to expose a 'join model' as that's more of an implementation thing (vs. a first-class schema IMO).

Most likely you have a `quiz` or something that's going to be exposed for a user (or `me`). That type will have some fields like `completedAt` (`isCompleted`, etc.).

Then, you want an artwork connection where the `node` in that connection is an `Artwork` (and not `QuizArtwork` which isn't really a first-class 'thing' if that makes sense) and the edge contains the 'join' fields (like `position` -> altho this might not be needed if the artworks are already ordered, `interactedAt`, etc.)

This is an example of that type of connection definition: https://github.com/artsy/metaphysics/blob/4f8f2f77bb2d73e958fd84bce517a6a161d05d51/src/schema/v2/userInquiredArtworks.ts#L43

That might be a better setup - I'm happy to help pair on that if curious!

Reply:
https://www.apollographql.com/blog/graphql/explaining-graphql-connections/ is the blog post that 🤯 on connections and how to think about them w/ keeping data on the edge, not really including the join model as the node, etc. if you are interested!